lib/index.js

```diff
@@ -372,6 +404,32 @@ function opt (option) {
   }
 };
 
+/* chan and rmchan sockopt methods. only relevant for subscription sockets */
+Socket.prototype.chan       = function(list){
+
+  var i = arguments.length;
+  while(i--) {
+    if (nn.Chan(this.binding, nn.NN_SUB_SUBSCRIBE, arguments[i]) !== -1) {
+      this.channels[arguments[i]] = true;
+    } else {
+      this.emit('error', new Error( er.op + er.chan + arguments[i] + '\n'));
+    }
+  };
+}
+
+Socket.prototype.rmchan     = function(){
+  var i = arguments.length;
+  while(i--) {
+    if (this.channels[arguments[i]]) {
+      if (nn.Chan(this.binding, nn.NN_SUB_UNSUBSCRIBE, arguments[i]) !== -1) {
+        delete this.channels[arguments[i]];
+      } else {
+        this.emit('error', new Error( er.op + er.rmchan + arguments[i] + '\n'));
```

Comment:
this error handler will throw an error.  Removed the definitions but did not replace the usages.